src/alexandria3k/json_source.py

```diff
@@ -0,0 +1,127 @@
+#
+# Alexandria3k Crossref bibliographic metadata processing
+# Copyright (C) 2023-2024 Evgenia Pampidi Diomidis Spinellis
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Cursors providing access to JSON data sources."""
+
+import abc
+
+from alexandria3k.data_source import ROWID_INDEX, ElementsCursor
+
+
+# Method names coming from apsw start with uppercase
+# pylint: disable=invalid-name
+
+
+class JSONElementsCursor(ElementsCursor):
```

Comment:
I can't see how this class is related to JSON. Maybe it should be in `data_source.py` with a name such as `NestedElementsCursor`? Given the class inheritance, please think hard regarding the appropriate names of all (new and existing) classes.  Feel free to send me a diagram and discuss this.  You can also send me a simple class diagram to discuss ideas.

Please also expand the docstring to explain what this cursor expects and what functionality it provided.